File: src/ts/src/facebook/FacebookGraphRequest.ts
export class FacebookGraphRequest {
    private _path: string;
    private readonly _parameters: { [key: string]: string };

    public constructor() {
        this._path = ``;
        this._parameters = {};
    }

    public setPath(path: string): this {
        this._path = path;
        return this;
    }

    public setParameter(key: string, value: string): this {
        this._parameters[key] = value;
        return this;
    }

    public toString(): string {
        return JSON.stringify({
            path: this._path,
            parameters: this._parameters,
        });
    }
}

File: src/cpp/ee/facebook/FacebookGraphRequest.cpp
//
//  FacebookGraphRequest.cpp
//  ee_x_mobile_facebook
//
//  Created by eps on 3/22/18.
//

#include "ee/facebook/FacebookGraphRequest.hpp"

#include <ee/nlohmann/json.hpp>

namespace ee {
namespace facebook {
using Self = GraphRequest;

Self& Self::setPath(const std::string& path) {
    path_ = path;
    return *this;
}

Self& Self::setParameter(const std::string& key, const std::string& value) {
    parameters_[key] = value;
    return *this;
}

std::string Self::toString() const {
    nlohmann::json json;
    json["path"] = path_;
    json["parameters"] = parameters_;
    return json.dump();
}
} // namespace facebook
} // namespace ee
